touchview/src/main/java/com/ortiz/touchview/TouchImageView.kt

```diff
@@ -823,34 +827,30 @@ class TouchImageView @JvmOverloads constructor(context: Context, attrs: Attribut
      * to the view's listener.
      */
     private inner class GestureListener : SimpleOnGestureListener() {
-        override fun onSingleTapConfirmed(e: MotionEvent): Boolean {
-            return if (doubleTapListener != null) {
-                doubleTapListener!!.onSingleTapConfirmed(e)
-            } else performClick()
+        override fun onSingleTapConfirmed(e: MotionEvent?): Boolean {
+            // Pass on to the OnDoubleTapListener if it is present, otherwise let the View handle the click.
+            return doubleTapListener?.onSingleTapConfirmed(e) ?: performClick()
         }
 
-        override fun onLongPress(e: MotionEvent) {
+        override fun onLongPress(e: MotionEvent?) {
             performLongClick()
         }
 
-        override fun onFling(e1: MotionEvent, e2: MotionEvent, velocityX: Float, velocityY: Float): Boolean {
-            if (fling != null) {
-                //
-                // If a previous fling is still active, it should be cancelled so that two flings
-                // are not run simultaenously.
-                //
-                fling!!.cancelFling()
-            }
+        override fun onFling(e1: MotionEvent?, e2: MotionEvent?, velocityX: Float, velocityY: Float): Boolean {
+            // If a previous fling is still active, it should be cancelled so that two flings
+            // are not run simultaneously.
+            fling?.cancelFling()
             fling = Fling(velocityX.toInt(), velocityY.toInt())
-            compatPostOnAnimation(fling!!)
+                    .also { compatPostOnAnimation(it) }
             return super.onFling(e1, e2, velocityX, velocityY)
         }
 
-        override fun onDoubleTap(e: MotionEvent): Boolean {
+        override fun onDoubleTap(e: MotionEvent?): Boolean {
             var consumed = false
-            if (isZoomEnabled) {
-                if (doubleTapListener != null) {
-                    consumed = doubleTapListener!!.onDoubleTap(e)
+            if (e != null && isZoomEnabled) {
+                val currentDoubleTapListener = doubleTapListener
+                if (currentDoubleTapListener != null) {
+                    consumed = currentDoubleTapListener.onDoubleTap(e)
```

Comment:
Why not get rid of variable `currentDoubleTapListener` at all and simplify it more with

                doubleTapListener?.let {
                    consumed = it.onDoubleTap(e)
                }

Reply:
Ah, nice one 👍